src/shared/clients/iotClient.ts

```diff
@@ -0,0 +1,702 @@
+            thing: new DefaultIotThing({
+                name: request.thingName,
+                region: this.regionCode,
+                arn: thingArn,
+            }),
+        }
+        getLogger().debug('CreateThing returned response: %O', response)
+        return response
+    }
+
+    /**
+     * Deletes an IoT Thing.
+     *
+     * @throws Error if there is an error calling IoT.
+     */
+    public async deleteThing(request: UpdateThingRequest): Promise<void> {
+        getLogger().debug('DeleteThing called with request: %O', request)
+        const iot = await this.createIot()
+
+        try {
+            await iot.deleteThing({ thingName: request.thingName }).promise()
+        } catch (e) {
+            getLogger().error('Failed to delete Thing: %O', e)
+            throw e
+        }
+
+        getLogger().debug('DeleteThing successful')
+    }
+
+    /**
+     * Lists all IoT certificates in account.
+     *
+     * @throws Error if there is an error calling IoT.
+     */
+    public async listCertificates(request: ListCertificatesRequest): Promise<ListCertificatesResponse> {
+        getLogger().debug('ListCertificates called with request: %O', request)
+        const iot = await this.createIot()
+
+        let iotCerts: Iot.Certificate[]
+        let nextMarker: Iot.Marker | undefined
+        try {
+            const output = await iot
+                .listCertificates({
+                    pageSize: request.pageSize ?? DEFAULT_MAX_THINGS,
+                    marker: request.marker,
+                    ascendingOrder: request.ascendingOrder,
+                })
+                .promise()
+            iotCerts = output.certificates ?? []
+            nextMarker = output.nextMarker
+        } catch (e) {
+            getLogger().error('Failed to retrieve certificates: %O', e)
+            throw e
+        }
+
+        const allCertsPromises: Promise<IotCertificate | undefined>[] = iotCerts.map(async iotCert => {
+            const certId = iotCert.certificateId
+            const certArn = iotCert.certificateArn
+            const certStatus = iotCert.status
+            const certDate = iotCert.creationDate
+            if (!certId || !certArn || !certStatus || !certDate) {
+                return undefined
+            }
+            return new DefaultIotCertificate({
+                arn: certArn,
+                id: certId,
+                activeStatus: certStatus,
+                creationDate: certDate,
+            })
+        })
+
+        const allCerts = await Promise.all(allCertsPromises)
+        const certs = _(allCerts)
+            .reject(cert => cert === undefined)
+            .map(cert => cert as IotCertificate)
+            .value()
+
+        const response: ListCertificatesResponse = { certificates: certs, nextMarker: nextMarker }
+        getLogger().debug('ListCertificates returned response: %O', response)
+        return { certificates: certs, nextMarker: nextMarker }
```

Comment:
Again, all of this should basically go away if returning the underlying types and returning a paginated iterator.